src/Module.Server/Modes/DTV/CrpgDTVTeamSelectComponent.cs

```diff
@@ -0,0 +1,330 @@
+    private readonly PeriodStatsHelper _periodStatsHelper;
+
+    /// <summary>
+    /// Players waiting to be assigned to a team when the cRPG balancer is enabled.
+    /// </summary>
+    private readonly HashSet<PlayerId> _playersWaitingForTeam;
+
+    public CrpgDTVTeamSelectComponent(MultiplayerWarmupComponent warmupComponent, MultiplayerRoundController? roundController)
+    {
+        _warmupComponent = warmupComponent;
+        _roundController = roundController;
+        _periodStatsHelper = new PeriodStatsHelper();
+        _playersWaitingForTeam = new HashSet<PlayerId>();
+    }
+#endif
+
+    public override void OnBehaviorInitialize()
+    {
+        base.OnBehaviorInitialize();
+
+#if CRPG_SERVER
+        _warmupComponent.OnWarmupEnded += OnWarmupEnded;
+        if (_roundController != null)
+        {
+            _roundController.OnPostRoundEnded += OnRoundEnded;
+        }
+    }
+
+    public override void OnRemoveBehavior()
+    {
+        base.OnRemoveBehavior();
+        if (_roundController != null)
+        {
+            _roundController.OnPostRoundEnded -= OnRoundEnded;
+        }
+
+        _warmupComponent.OnWarmupEnded -= OnWarmupEnded;
+    }
+
+    public void SetPlayerAgentsTeam()
+    {
+        Debug.Print("Setting player agents' team");
+
+        DTVGameMatch gameMatch = TeamsToGameMatch();
+        DTVGameMatch balancedGameMatch = MovePlayersToDefenderTeam(gameMatch);
+
+        Dictionary<int, Team> usersToMove = ResolveTeamMoves(current: gameMatch, target: balancedGameMatch);
+        var crpgNetworkPeers = GetCrpgNetworkPeers();
+        SendSwapNotification(usersToMove, crpgNetworkPeers);
+
+        foreach (var userToMove in usersToMove)
+        {
+            if (!crpgNetworkPeers.TryGetValue(userToMove.Key, out var networkPeer))
+            {
+                continue;
+            }
+
+            ChangeTeamServer(networkPeer, userToMove.Value);
+        }
+    }
+
+    protected override void AddRemoveMessageHandlers(GameNetwork.NetworkMessageHandlerRegistererContainer registerer)
+    {
+        registerer.Register<TeamChange>(HandleTeamChange);
+    }
+
+    private bool HandleTeamChange(NetworkCommunicator peer, TeamChange message)
+    {
+        if (message.Team == Mission.SpectatorTeam && !message.AutoAssign)
+        {
+            ChangeTeamServer(peer, message.Team);
+        }
+        else if (_warmupComponent.IsInWarmup)
+        {
+            AutoAssignTeam(peer);
+        }
+        else
+        {
+            var missionPeer = peer.GetComponent<MissionPeer>();
+            if (missionPeer is { Team: null })
+            {
+                // If the player just connected to the server, auto-assign their team so they have a chance
+                // to play the round.
+                AutoAssignTeam(peer);
+            }
+            else
+            {
+                _playersWaitingForTeam.Add(peer.VirtualPlayer.Id);
+            }
+        }
+
```

Comment:
This is what replace the balancer. It's completely enough to put people in teams.

Reply:
I've updated this to `ChangeTeamServer(peer, Mission.DefenderTeam);` for all occurences of `ChangeTeamServer()`. This is a temporary solution before disabling the option to join the other teams.

Reply:
This is fine. Having people land up in the only team no matter where they click (except for spectator) is correct IMO.